### PR TITLE
feat(sdk): SDK theming part 1 - black, bg-light, bg-dark, bg-black

### PR DESCRIPTION
**Pull request by deniskaber**
Relates to https://github.com/metabase/metabase/issues/42694
Also is a small part of https://www.notion.so/metabase/100-Align-approach-to-Mantine-styles-customization-1066f77c43694c36a6e90a3d2374b731

### Description

Migrate hardcoded colors used in emotion styles:

- black `${color("black")} ->var(--mb-color-bg-black)`
- bg-light `${color("bg-light")} -> var(--mb-color-bg-light)`
- bg-dark `${color("bg-dark")} -> var(--mb-color-bg-dark)`
- bg-black `${color("bg-black")} -> var(--mb-color-bg-black)`

### How to verify

1. CI should pass
2. Visual tests should return same results

### CODE REVIEW

File: frontend/src/metabase/nav/components/ReadOnlyBanner/ReadOnlyBanner.tsx [hunk 1]
```diff
@@ -5,8 +5,8 @@ import { Flex, Icon, Text } from "metabase/ui";
 export const ReadOnlyBanner = () => {
   return (
     <Flex py="0.75rem" px="1rem" bg="accent4" gap="md" align="center">
-      <Icon color="black" name="info_filled" />
-      <Text fw="bold" color="black">
+      <Icon color="text-dark" name="info_filled" />
+      <Text fw="bold" color="text-dark">
```
Comment (heypoom) [inline frontend/src/metabase/nav/components/ReadOnlyBanner/ReadOnlyBanner.tsx:9]: Black was `#2E353B` while text-dark is `#4C5773`, I think it's a slightly different shade?

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/icons/PublicEmbedIcon/PublicEmbedIcon.styled.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ export const PublicEmbedIconRoot = styled.svg<PublicEmbedIconRootProps>`
     .innerFill {
       stroke: ${disabled
         ? theme.fn.themeColor("text-light")
```
Comment (heypoom) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/icons/PublicEmbedIcon/PublicEmbedIcon.styled.tsx:16]: Does `var(--mb-color-text-light)` work here?

**Review by heypoom — APPROVED** [review]
Nice! Reviewed every file and there's 2 things I spotted, could you help double-check those?

Caveat: I haven't tested this change locally yet

File: frontend/src/metabase/css/core/colors.module.css [hunk 3]
```diff
@@ -11,9 +11,9 @@
   --mb-color-text-light: #949aab;
   --color-text-white: #fff;
   --mb-color-bg-black: #2e353b;
-  --color-bg-dark: #93a1ab;
+  --mb-color-bg-dark: #93a1ab;
```
Comment (WiNloSt) [inline frontend/src/metabase/css/core/colors.module.css:14]: For these new colors to work you need to put them in `<GlobalStyles />`. The reason is that we configured CSS module to remove these CSS custom properties, so their values are only accessible at run time, that means all variables here are only applicable for other CSS modules files, but they're accessible at run time i.e. styled components.

We could change the CSS configuration to always emit these CSS custom properties, or put these 2 colors in `GlobalStyles />` as mentioned.

You can see that The appearance color doesn't look the same anymore. The table header background has changed. This is the example where we couldn't access these colors in styled components.
#### This PR
image

#### Master
image

**Review by WiNloSt — DISMISSED** [review]

File: frontend/src/metabase/css/core/colors.module.css (hunk 3 above)
Comment (WiNloSt) [reply frontend/src/metabase/css/core/colors.module.css:14]: I see we're conditionally set the Post CSS custom property setting based on whether or not it's in the SDK context
https://github.com/metabase/metabase/blob/ea7bd55939d0dfd44b3e3e0605721e336c9429e4/postcss.config.js#L20-L24

What about just making it always `true`?

**Pull request merged**